core/src/main/scala/io/finch/ToService.scala

```diff
@@ -107,9 +110,16 @@ object ToService {
 
             val accept =
               if (opts.negotiateContentType) req.accept.map(a => Accept.fromString(a)) else Nil
-            out.map(oa =>
-              conformHttp(oa.toResponse(ntrA(accept), ntrE(accept)), req.version, opts)
-            ).run
+
+            val promise = Promise[Output[A]]
+            val io = effect.runAsync(out) {
+              case Left(t) => IO.pure(promise.setException(t))
+              case Right(value) => IO.pure(promise.setValue(value))
+            }
+            io.unsafeRunSync()
+            promise.map(oa =>
+              conformHttp(oa.convertToResponse(ntrA(accept).apply, ntrE(accept).apply), req.version, opts)
```

Comment:
I think we should move this map continuation under `Effect.unsafeRun*` and avoid bouncing through Future.